sbat.c

```diff
@@ -120,4 +123,161 @@ EFI_STATUS parse_sbat(char *sbat_base, size_t sbat_size, char *buffer,
 	return efi_status;
 }
 
+EFI_STATUS
+verify_sbat(struct sbat *sbat, list_t *sbat_entries)
+{
+	unsigned int i;
+	list_t *pos = NULL, *tmp = NULL;
+
+	for (i = 0; i < sbat->size; i++) {
+		struct sbat_entry *entry = entry = sbat->entries[i];
+		for_each_sbat_var_safe (pos, tmp, sbat_entries) {
+			struct sbat_var *sbat_var_entry;
+			sbat_var_entry = list_entry(pos, struct sbat_var, list);
+			if (strcmp(entry->component_name,
+			           sbat_var_entry->component_name) == 0) {
+				dprint(L"component %a has a matching SBAT variable entry, verifying\n", entry->component_name);
+				/* atoi returns zero for failed conversion, so essentially
+				   badly parsed component_generation will be treated as zero 
+				*/
+				UINT16 sbat_gen = atoi(entry->component_generation);
+				UINT16 sbat_var_gen = atoi(sbat_var_entry->component_generation);
+				if ( sbat_gen < sbat_var_gen ) {
+					dprint(L"component's %a generation: %d. Conflicts with SBAT variable generation %d\n",
```

Comment:
not sure I understand this debug message. What exactly conflicts means in this context ?

Maybe something like "component %a generation %d has been revoked by SBAT variable" or something like that?